drivers/sensor/ti_hdc20xx/ti_hdc20xx.c

```diff
@@ -0,0 +1,193 @@
+		/* x * 1000000 / 2^16 = x * 15625 / 2^10 */
+		val->val2 = ((tmp & 0xFFFF) * 15625U) >> 10;
+		break;
+	case SENSOR_CHAN_HUMIDITY:
+		/* val = 100 * sample / 2^16 */
+		tmp = data->rh_sample * TI_HDC20XX_RH_SCALE;
+		val->val1 = tmp >> 16;
+		/* x * 1000000 / 2^16 = x * 15625 / 2^10 */
+		val->val2 = ((tmp & 0xFFFF) * 15625U) >> 10;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api ti_hdc20xx_api_funcs = {
+	.sample_fetch = ti_hdc20xx_sample_fetch,
+	.channel_get = ti_hdc20xx_channel_get,
+};
+
+static int ti_hdc20xx_init(const struct device *dev)
+{
+	const struct ti_hdc20xx_config *config = dev->config;
+	uint16_t buf[2];
+	int rc;
+
+	if (!device_is_ready(config->bus)) {
+		LOG_ERR("I2C bus %s not ready", config->bus->name);
+		return -ENODEV;
+	}
+
+	/* manufacturer and device ID registers are consecutive, read them in the same burst */
+	rc = i2c_burst_read(config->bus, config->i2c_addr,
+			    TI_HDC20XX_REG_MANUFACTURER_ID, (uint8_t *)buf, sizeof(buf));
+	if (rc < 0) {
+		LOG_ERR("Failed to read manufacturer and device IDs");
+		return rc;
+	}
+
+	if (sys_le16_to_cpu(buf[0]) != TI_HDC20XX_MANUFACTURER_ID) {
+		LOG_ERR("Failed to get correct manufacturer ID");
+		return -EINVAL;
+	}
+	if (sys_le16_to_cpu(buf[1]) != TI_HDC20XX_DEVICE_ID) {
+		LOG_ERR("Unsupported device ID");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+/* Main instantiation macro */
+#define TI_HDC20XX_DEFINE(inst, compat)							\
+	static struct ti_hdc20xx_data ti_hdc20xx_data_##compat##inst;			\
+	static const struct ti_hdc20xx_config ti_hdc20xx_config_##compat##inst = {	\
+		.bus = DEVICE_DT_GET(DT_BUS(DT_INST(inst, compat))),			\
+		.i2c_addr = DT_REG_ADDR(DT_INST(inst, compat))				\
+	};										\
+	DEVICE_DT_DEFINE(DT_INST(inst, compat),						\
+			ti_hdc20xx_init,						\
+			NULL,								\
+			&ti_hdc20xx_data_##compat##inst,				\
+			&ti_hdc20xx_config_##compat##inst,				\
+			POST_KERNEL,							\
+			CONFIG_SENSOR_INIT_PRIORITY,					\
+			&ti_hdc20xx_api_funcs);
+
+/* Create the struct device for every status "okay" node in the devicetree. */
+#define TI_HDC20XX_FOREACH_STATUS_OKAY(compat, fn)	\
+	COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(compat),	\
+		    (UTIL_CAT(DT_FOREACH_OKAY_INST_,	\
+			      compat)(fn)),		\
+		    ())
+
+/*
+ * HDC2010 Low-Power Humidity and Temperature Digital Sensors
+ */
+#define TI_HDC2010_DEFINE(inst) TI_HDC20XX_DEFINE(inst, ti_hdc2010)
+TI_HDC20XX_FOREACH_STATUS_OKAY(ti_hdc2010, TI_HDC2010_DEFINE)
+
+/*
+ * HDC2021 High-Accuracy, Low-Power Humidity and Temperature Sensor
+ * With Assembly Protection Cover
+ */
+#define TI_HDC2021_DEFINE(inst) TI_HDC20XX_DEFINE(inst, ti_hdc2021)
+TI_HDC20XX_FOREACH_STATUS_OKAY(ti_hdc2021, TI_HDC2021_DEFINE)
+
+/*
+ * HDC2022 High-Accuracy, Low-Power Humidity and Temperature Sensor
+ * With IP67 Rated Water and Dust Protection Cover
+ */
+#define TI_HDC2022_DEFINE(inst) TI_HDC20XX_DEFINE(inst, ti_hdc2022)
```

Comment:
Looks like `dts/bindings/sensor/ti,hdc2022.yaml` is missing, otherwise OK

Reply:
Good catch, I have just fixed that